cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1642,6 +1642,55 @@ const char* convert_toUTF8(int64_t context, const char* value, int32_t value_len
   return value;
 }
 
+// Calculate the levenshtein distance between two string values
+FORCE_INLINE
+gdv_int32 levenshtein_utf8_utf8(int64_t context, const char* in1, int32_t in1_len,
```

Comment:
I feel there is a more space efficient algorithm than this as we are only interested in the edit distance and is not interested in reconstructing the edited input..Atleast the Java version uses this space efficient algorithm.

Is there any reason why we have to use this algorithm and not the one used by Java?

Reply:
Infact I feel more strongly now that we should use a memory efficient algorithm. What if the column width is 65K? we will need 4GB to run this alorithm. In fact Java version uses an algorithm that uses only 2n space where n is the smallest of the 2 strings.

Reply:
https://github.com/apache/commons-lang/blob/18a82dc4ae9c1095216b6cb288b3c5567ffd179b/src/main/java/org/apache/commons/lang3/StringUtils.java#L2336 has an implementation of levenshtein that use less memory.